tests/Functional/Driver/OCI8/ResultTest.php

```diff
@@ -0,0 +1,177 @@
+                . 'Error number: 2, Error String: oci_fetch_array.*/'
+            );
+        }
+
+        // Create a pipelined funtion that returns 10 rows more than the
+        // oci8 default prefetch
+        $this->createReturnTypeNeededForPipelinedFunction();
+        $expectedTotalRowCount = (int) ini_get('oci8.default_prefetch') + 10;
+        $this->createOrReplacePipelinedFunction($expectedTotalRowCount);
+
+        // Create a separate connection from that used to create/update the function
+        // This must be a different user with permissions to change the given function
+        $separateConnection = TestUtil::getPrivilegedConnection();
+
+        // Query the pipelined function to get initial dataset
+        $statement = $separateConnection->prepare(sprintf(
+            'SELECT * FROM TABLE(%s.test_oracle_fetch_failure())',
+            $this->connectionParams['user']
+        ));
+        $result    = $statement->execute();
+
+        // Access the first result to cause the first X rows to be prefetched
+        // as defined by oci8.default_prefetch (often 100 rows)
+        $firstNumber = $result->fetchOne();
+
+        if ($invalidateDataMidFetch) {
+            // Invalidate the original dataset by changing the pipelined function
+            // after the initial prefetch that caches locally the first X results
+            $this->createOrReplacePipelinedFunction($expectedTotalRowCount + 10);
+        }
+
+        while ($number = $result->fetchOne()) {
+            // Attempt to access all remaining rows from the original fetch
+            // The rows locally cached from the default prefetch will first be used
+            // but when the result attempts to get the remaining 10 rows beyond
+            // the first prefetch, nothing will be returned
+            //
+            // PHP oci8 oci_fetch_array will issue a PHP E_WARNING when the 2nd prefetch occurs
+            // oci_fetch_array(): ORA-04068: existing state of packages has been discarded
+            // ORA-04061: existing state of function "ROOT.TEST_ORACLE_FETCH_FAILURE" has been invalidated
+            // ORA-04065: not executed, altered or dropped function "ROOT.TEST_ORACLE_FETCH_FAILURE"
+            //
+            // If there was no issue, this should have returned rows totalling 10
+            // higher than the oci8 default prefetch
+            continue;
+        }
+
+        $this->assertEquals(
+            $expectedTotalRowCount,
+            $result->rowCount(),
+            sprintf(
+                'Expected to have %s total rows fetched but only found %s rows fetched',
+                $expectedTotalRowCount,
+                $result->rowCount()
+            )
+        );
+    }
+
+    public function dataProviderForTestTruncatedFetch(): Generator
+    {
+        yield 'it should return all rows if no data invalidation occurs'
+            => ['invalidateDataMidFetch' => false];
```

Comment:
The argument name may improve readability in the short term but it's not enforced at the testing framework level. I'd rather get rid of it and left the parameters as `[false]`.